snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -266,6 +267,36 @@ public Path getPath() {
     return snapshotsDirectory;
   }
 
+  @Override
+  public ActorFuture<Void> copySnapshot(
+      final PersistedSnapshot snapshot, final Path targetDirectory) {
+    final CompletableActorFuture<Void> result = new CompletableActorFuture<>();
+    actor.run(
+        () -> {
+          if (!Files.exists(snapshot.getPath())) {
+            result.completeExceptionally(
+                String.format(
+                    "Expected to copy snapshot %s to directory %s, but snapshot directory %s does not exits. Snapshot may have been deleted.",
+                    snapshot.getId(), targetDirectory, snapshot.getPath()),
+                new FileNotFoundException());
+          } else {
+            try {
+              LOGGER.info("Copying {} to {}", snapshot.getPath(), targetDirectory);
+              FileUtil.copySnapshot(targetDirectory, snapshot.getPath());
```

Comment:
Ya. I thought so too. But I didn't want to change a lot of things with this PR, so kept it as it is. I will change it since you also found it weird :)